rhizome/host/lib/vm_setup.rb

```diff
@@ -308,7 +308,13 @@ def cloudinit(unix_user, public_key, nics)
 
     write_user_data(unix_user, public_key)
 
-    r "mkdosfs -n CIDATA -C #{vp.q_cloudinit_img} 8192"
+    cloudinit_img_size = 8192 * 1024 # 8192 blocks * 1024 bytes/block
+
+    # Check if the file exists and has the correct size
+    if !File.exist?(vp.q_cloudinit_img) || File.size(vp.q_cloudinit_img) != cloudinit_img_size
```

Comment:
This also works, but one other option was adding `FileUtils.rm_rf(cloudinit_image)` before calling `mkdosfs`. I find reasoning about the rm_rf approach easier, because then I don't need to think about:
* What happens if `mcopy` is called with same arg twice?
* Is `mkdosfs` atomic? Does existence of file-size with a correct size mean it succeeded?

Also, if file exists and `File.size(vp.q_cloudinit_img) != cloudinit_img_size` is true, then `mkdosfs` will error-out anyway, and we need to remove the file anyway, so we can do it always.

Reply:
I agree, thanks for this.